atlasdb-config/src/main/java/com/palantir/atlasdb/http/TextDelegateDecoder.java

```diff
@@ -59,15 +64,23 @@ public TextDelegateDecoder(Decoder delegate) {
     }
 
     @Override
-    public Object decode(Response response, Type type) throws IOException, DecodeException, FeignException {
-        Collection<String> contentTypes = response.headers().get(HttpHeaders.CONTENT_TYPE);
-        // In the case of multiple content types, or an unknown content type, we'll use the delegate instead.
-        if (contentTypes != null
-                && contentTypes.size() == 1
-                && Iterables.getOnlyElement(contentTypes, "").equals(MediaType.TEXT_PLAIN)) {
-            return stringDecoder.decode(response, type);
+    public Object decode(Response response, Type type) throws IOException, FeignException {
+        try {
+            Collection<String> contentTypes = response.headers().entrySet().stream()
+                    .filter(entry -> entry.getKey().compareToIgnoreCase(HttpHeaders.CONTENT_TYPE) == 0)
+                    .map(Map.Entry::getValue)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+            // In the case of multiple content types, or an unknown content type, we'll use the delegate instead.
+            if (contentTypes != null
+                    && contentTypes.size() == 1
+                    && Iterables.getOnlyElement(contentTypes, "").equals(MediaType.TEXT_PLAIN)) {
+                return stringDecoder.decode(response, type);
```

Comment:
We should revert the changes to this file so that we just rebase in the fix from #1613